document/web/index.bs

```diff
@@ -0,0 +1,189 @@
+        text: Exported Function; url: #exported-function
+url:https://html.spec.whatwg.org/#cors-same-origin;text:CORS-same-origin;type:dfn;spec:HTML
+url:https://fetch.spec.whatwg.org/#concept-body-consume-body;text:consume body;type:dfn;spec:FETCH
+</pre>
+
+<pre class='link-defaults'>
+spec:ecma-262; type:exception; for:ECMAScript; text:TypeError
+</pre>
+
+This document builds off of the WebAssembly specification [[WEBASSEMBLY]] and the WebAssembly JavaScript embedding [[WASMJS]].
+
+<h2 id="streaming-modules">Streaming Module Compilation and Instantiation</h2>
+
+<pre class="idl">
+partial namespace WebAssembly {
+  Promise&lt;Module> compileStreaming(Promise&lt;Response> source);
+  Promise&lt;WebAssemblyInstantiatedSource> instantiateStreaming(
+      Promise&lt;Response> source, optional object importObject);
+};
+</pre>
+
+<div algorithm>
+The <dfn method for="WebAssembly">compileStreaming(source)</dfn> method, when invoked, returns the result of [=compile a potential WebAssembly response|compiling a potential WebAssembly response=] with |source|.
+</div>
+
+<div algorithm>
+The <dfn method for="WebAssembly">instantiateStreaming(source, importObject)</dfn> method, when invoked, performs the following steps:
+
+    1. Let |promiseOfModule| be the result of [=compile a potential WebAssembly response|compiling a potential WebAssembly response=] with |source|.
+    1. Return the result of [=instantiate a promise of a module|instantiating the promise of a module=] |promiseOfModule| with imports |importObject|.
+</div>
+
+<div algorithm>
+To <dfn>compile a potential WebAssembly response</dfn> with a promise of a {{Response}} |source|, perform the following steps:
+
+Note: This algorithm accepts a {{Response}} object, or a
+    promise for one, and compiles and instantiates the resulting bytes of the response. This compilation
+    can be performed in the background and in a streaming manner. If the {{Response}} is not
+    [=CORS-same-origin=], does not represent an [=ok status=], or does not match the
+    `` `application/wasm` `` MIME type, the returned promise will be rejected with a {{TypeError}}; if
+    compilation or instantiation fails, the returned promise will be rejected with a
+    {{CompileError}} or other relevant error type, depending on the cause of failure.
+
+
+    1. Let |returnValue| be [=a new promise=]
+    1. [=Upon fulfillment=] of |source| with value |unwrappedSource|:
+        1. Let |response| be |unwrappedSource|'s [=Response/response=].
+        1. Let |mimeType| be the result of [=extracting a MIME type=] from |response|'s [=response/header list=].
+        1. If |mimeType| is not `` `application/wasm` ``, reject |returnValue| with a {{TypeError}} and abort these substeps.
+
+        Note: extra parameters are not allowed, including the empty `` `application/wasm;` ``.
+
+        3. If |response| is not [=CORS-same-origin=], [=reject=] |returnValue| with a {{TypeError}} and abort these substeps.
```

Comment:
3 -> 1?

Reply:
This is done so the numbering continues after the note.